src/js/controller/controller.js

```diff
@@ -337,12 +338,26 @@ Object.assign(Controller.prototype, {
             if (model.get('playOnViewable')) {
                 if (viewable) {
                     _autoStart();
-                } else if (OS.mobile || (model.get('autoPause') && model.get('autoPause').viewability)) {
+                } else if (OS.mobile) {
                     _this.pause({ reason: 'autostart' });
                 }
             }
         }
 
+        if (_model.get('autoPause') && _model.get('autoPause').viewability) {
+            _model.set('pauseOnViewable', true);
+        }
+
+        function _checkPauseOnViewable() {
+            const adState = _getAdState();
+            const viewable = _model.get('viewable');
+            const isPaused = _model.get('state') === 'paused';
+
+            if (_model.get('pauseOnViewable') && !viewable && !adState && !isPaused) {
```

Comment:
Written this way your code will always do 4 model `get` calls. Write your logic so that the browser doesn't have to do more work when the logic is short-circuited:

```js
if (!viewable && _model.get('pauseOnViewable') && !_getAdState()) {
```

Also why bother checking for the paused state? You can call pause when already paused. What's going to happen? The only change would be to "pauseReason" on the model.

Reply:
We have to check if we're paused because if the user pauses by interaction and then scrolls out of view, the `pauseReason` gets overridden to `viewability` so when the user scrolls the player back into view, it resumes which we dont want.

Reply:
Hmm... maybe we shouldn't set a pauseReason if already paused - Because any call to pause could overwrite the initial pauseReason.